src/ng/rootScope.js

```diff
@@ -848,13 +849,14 @@ function $RootScopeProvider() {
 
         clearPhase();
 
-        while (postDigestQueue.length) {
+        while (postDigestQueuePosition < postDigestQueue.length) {
```

Comment:
Why not keep `postDigestQueuePosition` local and use a for loop as with `asyncQueue` above ?